### DISCUSSION TITLE
Prefix Window Title

### DISCUSSION THREAD

**Discussion by jdholtz** [kovidgoyal/kitty#6089] [Q&A] [answered]
Hello. I recently moved here from Alacritty and am loving Kitty in every way. I am currently using X11 and i3wm. How can I add a prefix to window titles (Ex: `~/Downloads` --> `kitty: ~/Downloads`). I tried to set it in `tab_title_template`, but that didn't have an effect on the window title, only the tab title. I searched through the issues as well, but couldn't find anything pertaining directly to my situation. Thanks!

**Comment by page-down** [comment, marked as answer]
This one is also on my list.
I would also like the title of the OS window to be customizable to show information that the child running in the kitty window does not know.

Currently there is no such option, but you can easily add such a prefix by monkey patching the kitty window title.
However this will be applied to all places such as tab titles, window title lists, etc.

> ... I am currently using X11 and i3wm. ...

Or maybe configuring `title_format` for i3wm would work?
```conf
for_window [class="kitty"] title_format "%class: %title"
```

**Reply by jdholtz** [reply]
The `for_window` rule works! Totally should've tried with something other than kitty. Thanks!

**Reply by jdholtz** [reply]
I have one more inconvenience with the window titles + shell integration. I want to execute a command like this: `kitty bash -c 'vim && bash'`. This command will open a vim session in a new window and return to the bash prompt when I exit. However, the new window has the title `bash` afterwards because that was the last command executed. Is there a way to reset Kitty to print the CWD again instead of the bash command?

**Comment by kovidgoyal** [comment]
If you do it like that, you disable automatic shell integration so
titles wont be updated. Either setup manual shell integration as
described in https://sw.kovidgoyal.net/kitty/shell-integration 
or run it like this
env COMMAND_TO_EXECUTE=vim kitty
and in your bashrc at the end add something like
[ -n "$COMMAND_TO_EXECUTE" ] && $COMMAND_TO_EXECUTE

**Reply by jdholtz** [reply]
That’s a very clever way to do it. Thanks!

**Reply by jdholtz** [reply]
Sorry to bother you @kovidgoyal, but this only actually works for programs that actively change the window title (e.g. vim). However, if I want to execute `man kitty`, the title will be 'bash' (You might have to put this in a script and execute the script instead of the command in the variable). Do you know of any workaround for this ~~besides using manual shell integration~~?

Edit: Using manual shell integration doesn't seem to work either

Edit 2: I can accomplish this by doing `xdotool getactivewindow set_window --name "command" && command`, which works in most cases (still displays 'bash' if I Ctrl+C for ssh), so I will opt for this option unless there is something better

**Reply by page-down** [reply]
> Using manual shell integration doesn't seem to work either

That's because the title is only updated after running a command from the prompt or when printing a shell prompt.

> ... man ..., the title will be 'bash' ...

Then just update the title before running man, just like kitty does in the shell integration script.

```bash
builtin printf "\e]2;%s\a" "my title"
```

> ... still displays 'bash' ...

Make sure that the bash you run after exiting with the shell integration manually enabled loads the relevant code.

Also you should be aware that if you change the title through the window manager, you are changing the title of the OS window.
You actually start a new kitty instance with a tab and a kitty window. If you update the title via the escape sequence, then it is the title of the kitty window that is updated, so the title of the OS window will change when you switch kitty windows or tabs.
(This doesn't seem to be relevant to your use case.)

**Reply by jdholtz** [reply]
> Make sure that the bash you run after exiting with the shell integration manually enabled loads the relevant code.

Oops. It wasn't running because I put it after the code that was executing when I used Ctrl+C.

Along with the printf command, everything works perfectly now! I really appreciate both of your guys help on this niche issue. This project has great development and user support. It is definitely a project I will model my projects off of.

**Comment by kovidgoyal** [comment]
Oh and in your bashrc also unset COMMAND_TO_EXECUTE (saving it in a local variable) before executing it so it is not inherited by child processes.
